### ISSUE TITLE
autoinstallation of extensions should happen on first run

### ISSUE THREAD

**Issue by boneskull** [appium/appium#17885]
Since a-somewhat-recent-version-of-`npm`, output by `postinstall` scripts is squelched by default.  Why?  People were complaining that `postinstall` scripts were begging for money or otherwise advertising.  

> To disable this behavior, an `npm` user can execute `npm install --foreground-scripts <pkg>`.

This directly affects `appium`, because `appium` has a `postinstall` script which automatically installs extensions _if_ specified by the user.  The output from this action will be suppressed, and may fail silently.  Worse, it may take a longer time than expected to install `appium`--without any feedback about why it's taking so long.  And there will be no feedback if the operation succeeded,  either.

To adapt to our _new reality_, if we want to support auto-installation of extensions, I don't see any other feasible solution than attempting to do so on first run of `appium`.

Thoughts?

**Comment by mykola-mokhnach** [comment]
I would just rather simply document this behavior. One can anyway install plugins later.

**Comment by jlipps** [comment]
surely there's at least a way if the postinstall fails to make sure that installation doesn't succeed?

**Comment by boneskull** [comment]
> surely there's at least a way if the postinstall fails to make sure that installation doesn't succeed?

If postinstall fails, installation will fail.  That's not the issue.  The issue is there's no positive feedback that the extensions got installed.

**Comment by subhashvkotra** [comment]
I could see 2 possible suggestions here:
1. Do the installation as pre-installation steps as suggested by @boneskull or
2. Show-up what all dependencies (as console output of appium installation) have failed and do not allow user to continue without installation
The 2nd option is not that great from user experience perspective.

**Comment by MohabMohie** [comment]
I came here looking for a solution for that exact problem... following the steps in the new user guide I wanted to setup my server in one fell swoop. the install command didn't give me any errors... upon investigation I found that it's not installing the plugins... and no matter what I try (double quotes in different locations, removing quotes, adding square brackets) nothing seems to work.
Any help?
![image](https://user-images.githubusercontent.com/19201898/217242879-1fe5bee4-d69f-4bfa-afe9-db6ff08f4a7d.png)
